File: library/src/main/java/com/nshmura/strictmodenotifier/ReportAdapter.java
package com.nshmura.strictmodenotifier;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;
import java.util.ArrayList;
import java.util.List;

class ReportAdapter extends BaseAdapter {
  List<StrictModeViolation> reports = new ArrayList<>();
  private StrictModeReportActivity reportActivity;

  public ReportAdapter(StrictModeReportActivity reportActivity) {
    this.reportActivity = reportActivity;
  }

  @Override public int getCount() {
    return reports.size();
  }

  @Override public StrictModeViolation getItem(int position) {
    return reports.get(position);
  }

  @Override public long getItemId(int position) {
    return position;
  }

  @Override public View getView(int position, View convertView, ViewGroup parent) {
    StrictModeViolation report = getItem(position);

    if (convertView == null) {
      convertView = LayoutInflater.from(reportActivity)
          .inflate(R.layout.strictmode_notifier_row, parent, false);
      convertView.setTag(new ViewHolder(convertView));
    }

    ViewHolder holder = (ViewHolder) convertView.getTag();
    holder.numberText.setText(
        parent.getContext().getString(R.string.strictmode_notifier_count, getCount() - position));
    holder.dateText.setText(report.getDateText(reportActivity));

    if (report.violationType != null) {
      holder.violationTypeText.setText(
          ViolationTypeInfo.convert(report.violationType).violationName());
    } else {
      holder.violationTypeText.setText(report.message);
    }

    return convertView;
  }

  public void addAll(List<StrictModeViolation> reports) {
    this.reports.addAll(reports);
  }

  public void clear() {
    this.reports.clear();
  }

  private class ViewHolder {
    final TextView numberText;
    final TextView violationTypeText;
    final TextView dateText;

    public ViewHolder(View convertView) {
      numberText = (TextView) convertView.findViewById(R.id.__number);
      violationTypeText = (TextView) convertView.findViewById(R.id.__violation_type);
      dateText = (TextView) convertView.findViewById(R.id.__date);
    }
  }
}

File: notifier/src/main/java/io/github/juanimoli/strictmode/notifier/ReportAdapter.kt
package io.github.juanimoli.strictmode.notifier

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import io.github.juanimoli.strictmode.notifier.commons.StrictModeViolation

internal class ReportAdapter(private val reportActivity: StrictModeReportActivity) : BaseAdapter() {
    private val reports: MutableList<StrictModeViolation> = ArrayList()

    override fun getCount(): Int {
        return reports.size
    }

    override fun getItem(position: Int): StrictModeViolation {
        return reports[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        var convertView = view

        if (convertView == null) {
            convertView = LayoutInflater.from(reportActivity)
                .inflate(R.layout.strictmode_notifier_row, parent, false)
            convertView.tag = ViewHolder(convertView)
        }

        val report = getItem(position)
        val holder = convertView!!.tag as ViewHolder

        holder.numberText.text =
            parent.context.getString(R.string.strictmode_notifier_count, count - position)
        holder.dateText.text = report.getDateText(reportActivity)
        if (report.violationType != null) {
            holder.violationTypeText.text =
                ViolationTypeInfo.convert(report.violationType).violationName()
        } else {
            holder.violationTypeText.text = report.message
        }

        return convertView
    }

    fun addAll(reports: List<StrictModeViolation>?) {
        this.reports.addAll(reports!!)
    }

    fun clear() {
        reports.clear()
    }

    private inner class ViewHolder(convertView: View) {
        val numberText: TextView = convertView.findViewById<View>(R.id.__number) as TextView
        val violationTypeText: TextView =
            convertView.findViewById<View>(R.id.__violation_type) as TextView
        val dateText: TextView = convertView.findViewById<View>(R.id.__date) as TextView
    }
}
